pkg/syncer/syncer.go

```diff
@@ -73,15 +74,63 @@ func (sc *SyncerConfig) ID() string {
 func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, importPollInterval time.Duration) error {
 	klog.Infof("Starting syncer for logical-cluster: %s, workload-cluster: %s", cfg.KCPClusterName, cfg.WorkloadClusterName)
 
-	upstreamConfig := rest.CopyConfig(cfg.UpstreamConfig)
-	upstreamConfig.UserAgent = "kcp#spec-syncer/v0.0.0"
-	downstreamConfig := rest.CopyConfig(cfg.DownstreamConfig)
-	downstreamConfig.UserAgent = "kcp#status-syncer/v0.0.0"
+	kcpVersion := version.Get().GitVersion
 
-	kcpClusterClient, err := kcpclient.NewClusterForConfig(upstreamConfig)
+	kcpClusterClient, err := kcpclient.NewClusterForConfig(rest.AddUserAgent(rest.CopyConfig(cfg.UpstreamConfig), "kcp#syncer/"+kcpVersion))
```

Comment:
please call the client-go default func as in the cli plugin. No need to invent a new scheme.